evm_arithmetization/src/generation/mod.rs

```diff
@@ -80,6 +84,22 @@ pub struct GenerationInputs {
     pub block_hashes: BlockHashes,
 }
 
+pub fn tx_hash(signed_txn: &[u8]) -> anyhow::Result<H256> {
+    let hash = keccak(signed_txn);
+    Ok(hash)
+}
```

Comment:
I don't think having this kind of function is useful. It:
- takes an opaque sequence of bytes, which may or may not be an actual encoded txn
- is simply a wrapper around `keccak` method over the same input arg
- returns a `Result` which will always be `Ok()`

Reply:
What if I inserted a call to `let _ = decode_as_txn(signed_txn)?` on line 88.

Reply:
I don't think that's the right way to address the problem. We shouldn't have to include unnecessary computation anywhere, especially as this won't be visible in the public API, which, if used in hot-spots, will have a non-negligible overhead due to this extra line.